rskj-core/src/main/java/co/rsk/peg/BridgeUtils.java

```diff
@@ -166,43 +166,109 @@ public static boolean isValidLockTx(BtcTransaction tx) {
+                return Coin.ZERO;
             }
 
             if (retiredFederationP2SHScript != null && scriptCorrectlySpendsTx(tx, index, retiredFederationP2SHScript)) {
-                return false;
+                return Coin.ZERO;
             }
         }
 
         Wallet federationsWallet = BridgeUtils.getFederationsNoSpendWallet(btcContext, activeFederations, false, null);
-        Coin valueSentToMe = tx.getValueSentToMe(federationsWallet);
 
+        return tx.getValueSentToMe(federationsWallet);
+    }
+
+    /**
+     * Verify if the value sent into the transaction is gratter or not the minimum, depending of the HF.
+     * @param valueSentToMe the value sent into the transaction in Coin
+     * @param bridgeConstants the Bridge constants
+     * @param txHash hash of the transaction (only to be used if the tx fails, to log info)
+     * @param activation calls different function to get minimum after or before Iris.
+     * @return true if this is a valid lock transaction
+     */
+    private static boolean isValidMinimumPegin(Coin valueSentToMe, BridgeConstants bridgeConstants, Sha256Hash txHash, ActivationConfig.ForBlock activation) {
         int valueSentToMeSignum = valueSentToMe.signum();
-        if (valueSentToMe.isLessThan(bridgeConstants.getMinimumLockTxValue())) {
-            logger.warn("[btctx:{}]Someone sent to the federation less than {} satoshis", tx.getHash(), bridgeConstants.getMinimumLockTxValue());
+        Coin minimumValue = activation.isActive(ConsensusRule.RSKIP219) ? bridgeConstants.getMinimumPeginTxValueInSatoshis() : bridgeConstants.getlegacyMinimumPeginTxValueInSatoshis();
+        if (valueSentToMe.isLessThan(minimumValue)) {
+            logger.warn("[btctx:{}]Someone sent to the federation less than {} satoshis", txHash, minimumValue);
         }
-        return (valueSentToMeSignum > 0 && !valueSentToMe.isLessThan(bridgeConstants.getMinimumLockTxValue()));
+        return (valueSentToMeSignum > 0 && !valueSentToMe.isLessThan(minimumValue));
+    }
+
+    /**
+     * It checks if the tx doesn't spend any of the federations' funds and if it sends more than
+     * the minimum ({@see BridgeConstants::getMinimumLockTxValue}) to any of the federations
+     * @param tx the BTC transaction to check
+     * @param activeFederations the active federations
+     * @param retiredFederationP2SHScript the retired federation P2SHScript. Could be {@code null}.
+     * @param btcContext the BTC Context
+     * @return true if this is a valid lock transaction
+     */
+    public static boolean isPegInTx(
+            BtcTransaction tx,
+            List<Federation> activeFederations,
+            Script retiredFederationP2SHScript,
+            Context btcContext
+    ) {
+        Coin valueSentToMe = getValueSentInPeginTx(tx, activeFederations, retiredFederationP2SHScript, btcContext);
+
+        int valueSentToMeSignum = valueSentToMe.signum();
+        return (valueSentToMeSignum > 0);
```

Comment:
```suggestion
        return valueSentToMe.isPositive();
```
It does the same thing and it think it's clearer